### PR TITLE
feat(nextjs): support next 11

### LINKED ISSUE #6019: Next.js v11 support
## Description
Next.js v11 has been released. I want nx to support v11.
Also, next lint is officially available. I want nx to provide a way to incorporate it.

https://github.com/vercel/next.js/releases/tag/v11.0.0

https://nextjs.org/blog/next-11

### PR DESCRIPTION
**Pull request by kirjai**
An important svg import issue has been fixed by next 11.0.1 <3 

## Related Issue(s)

Fixes #

### CODE REVIEW

File: e2e/next/src/next.test.ts [hunk 1]
```diff
@@ -318,7 +319,7 @@ describe('Next.js Applications', () => {
     });
   }, 120000);
 
-  it('should build with public folder', async () => {
+  it.only('should build with public folder', async () => {
```
Comment (gperdomor) [inline e2e/next/src/next.test.ts:322]: this is ok?

Comment (kirjai) [reply e2e/next/src/next.test.ts]: still testing some things out, still in draft 😀

File: packages/next/migrations.json [hunk 2]
```diff
@@ -119,6 +125,19 @@
...
       }
+    },
+    "12.6.0": {
+      "version": "12.6.0-beta.0",
+      "packages": {
+        "next": {
+          "version": "11.0.1",
+          "alwaysAddToPackageJson": false
+        },
+        "eslint-config-next": {
```
Comment (jaysoo) [inline packages/next/migrations.json:136]: As discussed, let's move it to the actual migration so we can install it as a dev dependency.

**Review by jaysoo — CHANGES_REQUESTED** [review]

File: packages/next/migrations.json (hunk 2 above)
Comment (kirjai) [reply packages/next/migrations.json:136]: i'm hoping that https://github.com/nrwl/nx/pull/6287 will be merged soon, and then i can just use that, tbh. but if that PR will take too long, i'll continue with this PR without it

Comment (kirjai) [reply packages/next/migrations.json]: updated PR to not rely on #6287 anymore, so we can merge this PR sooner rather than later

**Pull request merged**